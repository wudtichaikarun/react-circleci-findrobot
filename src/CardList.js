import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  // if (true) {
  //   throw new Error('Nooo...');
  // }

  return (
    <div>
      {robots.map(users => (
        <Card
          key={users.id}
          id={users.id}
          name={users.name}
          email={users.email}
        />
      ))}
    </div>
  );
};

export default CardList;
