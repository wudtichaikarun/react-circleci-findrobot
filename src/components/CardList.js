import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  return (
    <div>
      {robots.map(users => (
        <Card
          key={users.id}
          id={users.id}
          name={users.name}
          email={users.email}
        />
      ))}
    </div>
  );
};

export default CardList;
