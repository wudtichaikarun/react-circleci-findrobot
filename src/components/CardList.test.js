import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'Romantic',
      email: 'Ro@gamil.com'
    }
  ];
  expect(
    shallow(
      <CardList robots={mockRobots} />
    )
  ).toMatchSnapshot();
});
