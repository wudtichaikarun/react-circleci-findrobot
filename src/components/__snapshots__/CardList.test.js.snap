// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "Ro@gamil.com",
              "id": 1,
              "name": "Romantic",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="Ro@gamil.com"
          id={1}
          name="Romantic"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "Ro@gamil.com",
          "id": 1,
          "name": "Romantic",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="Ro@gamil.com"
            id={1}
            name="Romantic"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "Ro@gamil.com",
            "id": 1,
            "name": "Romantic",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
